src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react';
+import styles from './Grid.st.css';
+import {
+  generateListClassId,
+  getMediaQueries,
+  getMinWidthByCardType,
+  itemsPerRowWidth,
+} from './GridUtils';
+import {
+  TPAComponentsConsumer,
+  TPAComponentsContext,
+} from '../TPAComponentsConfig';
+import style from '../Button/Button.st.css';
+
+export const ROW_GAP = 32;
+export const COLUMN_GAP = 32;
+export const MOBILE_ROW_GAP = 20;
+export const MOBILE_COLUMN_GAP = 20;
+export const COLUMS_SPAN = 1;
+export const ROW_SPAN = 1;
+
+export interface IGridItem {
+  item: React.ReactNode;
+  key?: string | number;
+  colSpan?: number;
+  rowSpan?: number;
+}
+
+export interface GridProps {
+  withDivider: boolean;
+  items: IGridItem[];
+  maxColumns?: number;
+  minColumnWidth?: number;
+  width?: number;
+  rowGap?: number;
+  columnGap?: number;
+  dividerWidth?: string | number;
+}
+
+interface DefaultProps {
+  withDivider: boolean;
+  items: IGridItem[];
+  maxColumns: number;
+  width: number;
+  dividerWidth: string;
+}
+
+export class Grid extends React.Component<GridProps> {
+  static contextType = TPAComponentsContext;
+  static displayName = 'Grid';
+  static defaultProps: DefaultProps = {
+    withDivider: false,
+    items: [],
+    maxColumns: 1,
+    width: 0,
+    dividerWidth: '1px',
+  };
+
+  render() {
+    return (
+      <TPAComponentsConsumer>
+        {({ mobile }) => {
+          const { withDivider, width, items, ...rest } = this.props;
+          const rowGap =
+            this.props.rowGap || (mobile ? MOBILE_ROW_GAP : ROW_GAP);
```

Comment:
Correct me if i'm wrong, but this means I can't set a `rowGap` of `0`, right?

Reply:
correct